checker/src/main/java/org/checkerframework/checker/calledmethods/builder/LombokSupport.java

```diff
@@ -0,0 +1,202 @@
+import javax.lang.model.element.Element;
+import javax.lang.model.element.ElementKind;
+import javax.lang.model.element.ExecutableElement;
+import javax.lang.model.element.Name;
+import javax.lang.model.element.TypeElement;
+import org.checkerframework.checker.calledmethods.CalledMethodsAnnotatedTypeFactory;
+import org.checkerframework.framework.type.AnnotatedTypeMirror;
+import org.checkerframework.framework.type.AnnotatedTypeMirror.AnnotatedExecutableType;
+import org.checkerframework.javacutil.AnnotationUtils;
+import org.checkerframework.javacutil.ElementUtils;
+
+/**
+ * Lombok Support for the Called Methods Checker, which adds CalledMethods annotations to the code
+ * generated by Lombok.
+ */
+public class LombokSupport implements BuilderFrameworkSupport {
+
+    /** The type factory. */
+    private CalledMethodsAnnotatedTypeFactory atypeFactory;
+
+    /**
+     * Create a new LombokSupport.
+     *
+     * @param atypeFactory the typechecker's type factory
+     */
+    public LombokSupport(CalledMethodsAnnotatedTypeFactory atypeFactory) {
+        this.atypeFactory = atypeFactory;
+    }
+
+    // The list is copied from lombok.core.handlers.HandlerUtil. The list cannot be used from that
+    // class directly because Lombok does not provide class files for its own implementation, to
+    // prevent itself from being accidentally added to clients' compile classpaths. This design
+    // decision means that it is impossible to depend directly on Lombok internals.
+    /** The list of annotations that Lombok treats as non-null. */
+    public static final List<String> NONNULL_ANNOTATIONS =
+            Collections.unmodifiableList(
+                    Arrays.asList(
+                            "android.annotation.NonNull",
+                            "android.support.annotation.NonNull",
+                            "com.sun.istack.internal.NotNull",
+                            "edu.umd.cs.findbugs.annotations.NonNull",
+                            "javax.annotation.Nonnull",
+                            // "javax.validation.constraints.NotNull", // The field might contain a
+                            // null value until it is persisted.
+                            "lombok.NonNull",
+                            "org.checkerframework.checker.nullness.qual.NonNull",
+                            "org.eclipse.jdt.annotation.NonNull",
+                            "org.eclipse.jgit.annotations.NonNull",
+                            "org.jetbrains.annotations.NotNull",
+                            "org.jmlspecs.annotation.NonNull",
+                            "org.netbeans.api.annotations.common.NonNull",
+                            "org.springframework.lang.NonNull"));
+
+    /**
+     * A map from elements that have a lombok.Builder.Default annotation to the simple property name
+     * that should be treated as defaulted.
+     *
+     * <p>This cache is kept so that when declarationFromElement doesn't work, defaults are still
```

Comment:
I added a comment, copied below:

>     * <p>This cache is kept because the usual method for checking that an element has been defaulted
>     * (calling declarationFromElement and examining the resulting VariableTree) only works if a corresponding
>     * Tree is available (for code that is only available as bytecode, no such Tree is available and that
>     * method returns null). See the code in {@link #getLombokRequiredProperties(Element)} that handles
>     * fields.